File: services/dbservice/src/main/java/com/torin/dbService/dto/LogEntryDto.java
package com.torin.dbService.dto;

import java.time.Instant;
import java.time.ZoneOffset;
import java.time.format.DateTimeFormatter;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
public class LogEntryDto {
    @JsonProperty("handler_id")
    private String handlerId;
    @JsonProperty("timestamp")
    private Instant timestamp;

    @JsonProperty("message")
    public String message;
    @JsonProperty("level")
    public String level;

    @JsonProperty("formatter_timestamp")
    public String getFormatterTimestamp() {
        if (timestamp == null) {
            return null;
        }

        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");

        return timestamp
                .atZone(ZoneOffset.UTC)
                .format(formatter);
    }

    public static LogEntryDto create(String handlerId, String message, String level, Instant timestamp) {
        if (timestamp == null) {
            timestamp = Instant.now();
        }
        return new LogEntryDto(handlerId, timestamp, message, level);
    }
}

File: services/telegramservice/DTOs/DTO/LogEntryDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Newtonsoft.Json;

namespace DTOs.DTO
{
    public class LogEntryDTO
    {
        [JsonProperty("handler_id")]
        public string HandlerId { get; set; } = string.Empty;
        [JsonProperty("timestamp")]
        public DateTime Timestamp { get; set; }
        [JsonProperty("formatter_timestamp")]
        public string? FormatterTimestamp { get; set; }
        [JsonProperty("message")]
        public string Message { get; set; } = string.Empty;
        [JsonProperty("level")]
        public string Level { get; set; } = string.Empty;
    }
}